### PR TITLE
Util/unordered

### PR DESCRIPTION
**Pull request by AlemSnyder**
Here you go Nino. Unordered maps and sets.

### CODE REVIEW

File: src/world/biome.cpp [hunk 1]
```diff
@@ -381,16 +381,17 @@ Biome::init_materials_(const Json::Value& material_data) {
     return out;
 }
 
-std::map<ColorInt, std::pair<const Material*, ColorId>>
+std::unordered_map<ColorInt, std::pair<const Material*, ColorId>>
 Biome::get_colors_inverse_map() const {
-    std::map<ColorInt, std::pair<const Material*, ColorId>> materials_inverse;
+    std::unordered_map<ColorInt, std::pair<const Material*, ColorId>> materials_inverse;
```
Comment (egelja) [inline src/world/biome.cpp:386]: instead of a pair, amke a struct

File: src/gui/render/graphics_shaders/program_handler.hpp
Comment (egelja) [inline src/gui/render/graphics_shaders/program_handler.hpp:1]: why commented out code

**Review by egelja — CHANGES_REQUESTED** [review]

File: src/gui/render/graphics_shaders/program_handler.hpp
Comment (AlemSnyder) [reply src/gui/render/graphics_shaders/program_handler.hpp:1]: These were my hashes. I couldn't get them to work. I don't know why. Could you take a look?

File: src/world/terrain/material.hpp [hunk 2]
```diff
@@ -65,6 +65,22 @@ struct Material {
...
+    ColorId color;
+
+    inline MaterialColor(const MaterialColor& other) : material(other.material), color(other.color){
+    }
+
+
+    inline MaterialColor(MaterialColor&& other) : material(other.material), color(other.color){
+    }
+
+    inline MaterialColor(const Material& mat, ColorId col) : material(mat), color(col) {}
```
Comment (egelja) [inline src/world/terrain/material.hpp:79]: Structs don't need these they are auto generated

**Review by egelja — APPROVED** [review]
LGTM, I'd recommend fixing the constructor thing just for clarity

File: src/world/terrain/material.hpp (hunk 2 above)
Comment (AlemSnyder) [reply src/world/terrain/material.hpp]: I don't know but this breaks things.

Comment (egelja) [reply src/world/terrain/material.hpp]: You have to change the call to the ctor, left a comment specifying how

File: src/world/biome.cpp [hunk 3]
```diff
@@ -381,17 +381,12 @@ Biome::init_materials_(const Json::Value& material_data) {
...
+    std::unordered_map<ColorInt, MaterialColor> materials_inverse;
     for (const auto& element : materials_) {
         for (ColorId color_id = 0; color_id < element.second.color.size(); color_id++) {
-            materials_inverse.insert(
-                std::map<ColorInt, std::pair<const Material*, ColorId>>::value_type(
-                    element.second.color.at(color_id).second,
-                    std::make_pair(&element.second, color_id)
-                )
-            );
+            materials_inverse.emplace(color_id, MaterialColor(element.second, color_id));
```
Comment (egelja) [inline src/world/biome.cpp]:
```suggestion
            materials_inverse.emplace(color_id, MaterialColor{element.second, color_id});
```

Comment (AlemSnyder) [reply src/world/biome.cpp]:
```sh
/home/alem/Documents/code_porjects/Fun-Game/src/world/biome.cpp:390:65: error: no matching function for call to ‘terrain::MaterialColor::MaterialColor(<brace-enclosed initializer list>)’
  390 |                 color_id, MaterialColor{element.second, color_id}
      |                                                                 ^
```

Comment (egelja) [reply src/world/biome.cpp]: You need to delete the other ctors from the struct

**Pull request merged**